### PR TITLE
MM-11567: Autocomplete search in: for DMs and GMs

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Autocomplete search in: for DMs and GMs

#### Ticket Link
[MM-11567](https://mattermost.atlassian.net/browse/MM-11567)

Depends on mattermost/mattermost-server#9430

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Device Information
This PR was tested on: [One Plus 5, Android 8.1]

### CODE REVIEW

**Comment by hmhealey** [comment]
Does this need a version check to prevent the autocomplete showing up for older servers? What happens if you try to search in a DM or GM channel on an older server?

**Comment by jespino** [comment]
Older servers wouldn't be a problem, the problem could be new server with old native app.

**Comment by hmhealey** [comment]
Does it show the autocomplete for DM channels on when you have a new app on an old server though? We don't want to offer the user to use something that doesn't work.

New server with old app should be fine as long as the new server isn't breaking the existing API

**Comment by jespino** [comment]
If you have the new application with an old server, you use the old endpoint, so It will work as expected for old server. If you have an old RN app, with the new server, you still use the old endpoint (working as expected). I think we are covered here.

**Comment by hmhealey** [comment]
It won't cause any errors, but the user will be shown a new feature that doesn't work because it won't return any results. Like this is what it looks like when I try to use it on an older server
screen shot 2018-09-26 at 10 20 46 am
screen shot 2018-09-26 at 10 20 34 am

**Comment by jespino** [comment]
before this, the autocomplete for search list was autocompleting with all public channels, so, you had this kind of things easily. But, I can change that to detect old servers and do not complete that.

**Comment by jespino** [comment]
@hmhealey Now it only shows direct messages for mattermost server V5.4 or greater

File: app/components/autocomplete/channel_mention_item/index.js [hunk 1]
```diff
@@ -6,15 +6,23 @@ import {connect} from 'react-redux';
...
 
 import ChannelMentionItem from './channel_mention_item';
 
 function mapStateToProps(state, ownProps) {
     const channel = getChannel(state, ownProps.channelId);
+    let displayName = channel.display_name;
+
+    // Bypassing the channel display name generation in DMs and GMs
+    if (channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL) {
+        displayName = state.entities.channels.channels[ownProps.channelId].display_name;
```
Comment (sudheerDev) [inline app/components/autocomplete/channel_mention_item/index.js]: Could you use a selector to get channel here? so if we ever change state object we don't run into issues?

Comment (enahum) [reply app/components/autocomplete/channel_mention_item/index.js]: 2/5 to use a selector

**Comment by jespino** [comment]
@enahum In this case I'm explicitly avoiding to get the channel from the selector, because i receive the channel display_name from the server populated (for autocomplete search results), if I use the getChannel selector, it populates automatically the display_name for the DirectMessage channel.

**Comment by enahum** [comment]
@jespino can you create a specific selector for that?

**Pull request merged**